common/lib/fdt.c

```diff
@@ -0,0 +1,33 @@
+#if !defined(__i386__) && !defined(__x86_64__)
+
+#include <stdint.h>
+#include <stddef.h>
+#include <libfdt/libfdt.h>
+
+static int fdt_get_or_add_chosen_node(void *fdt) {
+    int offset = fdt_subnode_offset(fdt, 0, "chosen");
+
+    if (offset == -FDT_ERR_NOTFOUND) {
+        offset = fdt_add_subnode(fdt, 0, "chosen");
+    }
+
+    return offset;
+}
+
+int fdt_set_chosen_string(void *fdt, const char *name, const char *value) {
+    int chosen_offset = fdt_get_or_add_chosen_node(fdt);
+    if (chosen_offset < 0)
```

Comment:
One last thing, style still, could you use braces even for single line ifs? Thank you! (This is done for all new code, and at some point I should go over the whole codebase to change it)

Reply:
Okay. It is helpful to have explicit style documentation and style checker/formatter configuration (like `.clang-format`)

Reply:
I do not like it.